src/utils/s3-utils.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { GetObjectCommand, ListObjectsV2Command } from '@aws-sdk/client-s3';
+
+export async function getObjectKeysUsingPrefix(s3Client, bucketName, prefix, log) {
+  const objectKeys = [];
+  try {
+    const params = {
+      Bucket: bucketName,
+      Prefix: prefix,
+      MaxKeys: 1000,
+    };
+    const data = await s3Client.send(new ListObjectsV2Command(params));
+    data?.Contents?.forEach((obj) => {
+      objectKeys.push(obj.Key);
+    });
+    log.info(`Fetched ${objectKeys.length} keys from S3 for bucket ${bucketName} and prefix ${prefix}`);
```

Comment:
If more than 1000 objects are expected, you should handle pagination by repeatedly calling `ListObjectsV2` with `ContinuationToken`

```

let continuationToken = null;
do {
  const params = {
    Bucket: bucketName,
    Prefix: prefix,
    MaxKeys: 1000,
    ContinuationToken: continuationToken,
  };
  const data = await s3Client.send(new ListObjectsV2Command(params));
  data?.Contents?.forEach((obj) => {
    objectKeys.push(obj.Key);
  });
  continuationToken = data.IsTruncated ? data.NextContinuationToken : null;
} while (continuationToken);
```

Reply:
The `ListObject` command doesn't returns the object contents but just the objects metadata like key etc so memory wise it is not much. The listObjectsV2 API has a limit of 1,000 objects per request, and since we are scraping top-pages only which are expected to be < 200, we should be good here in my opinion. Let me know your thoughts.